src/main/java/software/aws/neptune/common/gremlindatamodel/resultset/ResultSetGetTypeInfo.java

```diff
@@ -0,0 +1,224 @@
+        ORDERED_COLUMNS.add(MINIMUM_SCALE);
+        ORDERED_COLUMNS.add(MAXIMUM_SCALE);
+        ORDERED_COLUMNS.add(SQL_DATA_TYPE);
+        ORDERED_COLUMNS.add(SQL_DATETIME_SUB);
+        ORDERED_COLUMNS.add(NUM_PREC_RADIX);
+
+        COLUMN_TYPE_MAP.put(TYPE_NAME, String.class);
+        COLUMN_TYPE_MAP.put(DATA_TYPE, Types.class);
+        COLUMN_TYPE_MAP.put(PRECISION, Integer.class);
+        COLUMN_TYPE_MAP.put(LITERAL_PREFIX, String.class);
+        COLUMN_TYPE_MAP.put(LITERAL_SUFFIX, String.class);
+        COLUMN_TYPE_MAP.put(CREATE_PARAMS, String.class);
+        COLUMN_TYPE_MAP.put(NULLABLE, Short.class);
+        COLUMN_TYPE_MAP.put(CASE_SENSITIVE, Boolean.class);
+        COLUMN_TYPE_MAP.put(SEARCHABLE, Short.class);
+        COLUMN_TYPE_MAP.put(UNSIGNED_ATTRIBUTE, Boolean.class);
+        COLUMN_TYPE_MAP.put(FIXED_PREC_SCALE, Boolean.class);
+        COLUMN_TYPE_MAP.put(AUTO_INCREMENT, Boolean.class);
+        COLUMN_TYPE_MAP.put(LOCAL_TYPE_NAME, String.class);
+        COLUMN_TYPE_MAP.put(MINIMUM_SCALE, Short.class);
+        COLUMN_TYPE_MAP.put(MAXIMUM_SCALE, Short.class);
+        COLUMN_TYPE_MAP.put(SQL_DATA_TYPE, Integer.class);
+        COLUMN_TYPE_MAP.put(SQL_DATETIME_SUB, Integer.class);
+        COLUMN_TYPE_MAP.put(NUM_PREC_RADIX, Integer.class);
+    }
+
+    protected static void populateConstants(List<Map<String, Object>> typeInfo) {
+        for (Map<String, Object> info : typeInfo) {
+            info.put("CREATE_PARAMS", null);
+            info.put("NULLABLE", typeNullable);
+            info.put("SEARCHABLE", typeSearchable);
+            info.put("FIXED_PREC_SCALE", false);
+            info.put("AUTO_INCREMENT", false);
+            info.put("LOCAL_TYPE_NAME", null);
+            info.put("SQL_DATA_TYPE", null);
+            info.put("SQL_DATETIME_SUB", null);
+        }
+    }
+
+    protected static void putInfo(List<Map<String, Object>> typeInfo, String typeName, int dataType, boolean isText,
+                                  boolean isNumeric, boolean isUnsignedAttribute) {
+        final Map<String, Object> info = new HashMap<>();
+        info.put(TYPE_NAME, typeName);
+        info.put("DATA_TYPE", dataType);
+        info.put("PRECISION", getTypePrecision(dataType));
+        info.putIfAbsent("UNSIGNED_ATTRIBUTE", isUnsignedAttribute);
+
+        if (isText) {
+            info.put("LITERAL_PREFIX", "'");
+            info.put("LITERAL_SUFFIX", "'");
+            info.put("CASE_SENSITIVE", true);
+        } else {
+            info.put("LITERAL_PREFIX", null);
+            info.put("LITERAL_SUFFIX", null);
+            info.put("CASE_SENSITIVE", false);
+        }
+
+        if (isNumeric) {
+            info.put("MINIMUM_SCALE", 0);
+            info.put("MAXIMUM_SCALE", 0);
+            info.put("NUM_PREC_RADIX", 10);
+        } else {
+            info.put("MINIMUM_SCALE", null);
+            info.put("MAXIMUM_SCALE", null);
+            info.put("NUM_PREC_RADIX", null); 
+        }
```

Comment:
Nit: Might be nice to condense this with ternary operator.
```suggestion
        info.put("LITERAL_PREFIX", isText ? "'" : null);
        info.put("LITERAL_SUFFIX", isText ? "'" : null);
        info.put("CASE_SENSITIVE", isText);
        info.put("MINIMUM_SCALE", isNumeric ? 0 : null);
        info.put("MAXIMUM_SCALE", isNumeric ? 0 : null);
        info.put("NUM_PREC_RADIX", isNumeric ? 10 : null); 
```